### PR TITLE
Add sanity check to RPNBuilderFunctionTreeNode

### PR DESCRIPTION
**Pull request by rschu1ze**
See Bug: #52632

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 77e7f71e02bb71e2f571d0897c9dc4fb684d6db3 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55915/77e7f71e02bb71e2f571d0897c9dc4fb684d6db3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6605890556">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55915/77e7f71e02bb71e2f571d0897c9dc4fb684d6db3/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/MergeTree/RPNBuilder.cpp [hunk 1]
```diff
@@ -396,6 +396,12 @@ size_t RPNBuilderFunctionTreeNode::getArgumentsSize() const
 
 RPNBuilderTreeNode RPNBuilderFunctionTreeNode::getArgumentAt(size_t index) const
 {
+    size_t total_arguments = getArgumentsSize();
```
Comment (devcrafter) [inline src/Storages/MergeTree/RPNBuilder.cpp:399]:
```suggestion
    const size_t total_arguments = getArgumentsSize();
```

File: src/Storages/MergeTree/RPNBuilder.cpp [hunk 2]
```diff
@@ -396,6 +396,12 @@ size_t RPNBuilderFunctionTreeNode::getArgumentsSize() const
...
+    if (index >= total_arguments) /// Bug #52632
```
Comment (devcrafter) [inline src/Storages/MergeTree/RPNBuilder.cpp:400]:
```suggestion
    if (unlikely(index >= total_arguments)) /// Bug #52632
```

**Review by devcrafter — APPROVED** [review]

File: src/Storages/MergeTree/RPNBuilder.cpp (hunk 2 above)
Comment (rschu1ze) [reply src/Storages/MergeTree/RPNBuilder.cpp:400]: Not an unreasonable request per se, but none of the code is performance-critical (e.g. in a hot loop). I'd prefer to not clutter the code and keep it as is.

Btw., the "native" alternative to likely/unlikely would be `[[likely]]` / `[[unlikely]]` ([C++20 docs](https://en.cppreference.com/w/cpp/language/attributes/likely)).

**Pull request merged**